clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/LeaderFollowerStoreIngestionTask.java

```diff
@@ -2285,7 +2285,7 @@ protected DelegateConsumerRecordResult delegateConsumerRecord(
             consumerRecord,
             isEndOfPushReceived,
             partitionConsumptionState);
-        versionedDIVStats.recordLeaderDIVCompletionTime(
+        versionedDIVStats.recordLeaderProcessToDIV(
```

Comment:
Thanks for iterating on this metric name as well.

So the scope of this metric is to perform various metadata checks and bookkeeping, beginning from the start of taking a record out of the consumed batch, up until just before the point where we'll do something more significant with it (e.g. pass it to the drainer for local persistence or to the producer). The one thing which may take a significant time during this process appears to be the `SIT::waitReadyToProcessRecord` function, which will block until the availability of any schema that has never been seen yet, and then attempt to use that schema to deserialize and potentially decompress the incoming payload. If this is the phenomenon we're interested in, then I wonder if we should add a metric just for this, rather than have a metric which covers several loosely defined steps...